### ISSUE TITLE
[250502_OhLoRA_v2] StyleGAN 구조 변경

### ISSUE THREAD

**Issue by WannaBeSuperteur** [WannaBeSuperteur/AI_Projects#1]
## Project README

- [Project README](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_05_02_OhLoRA_v2/README.md)
- [Schedule Table](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_05_02_OhLoRA_v2/README.md#3-%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8-%EC%9D%BC%EC%A0%95)

## Task List

- [x] 논문 탐독 (05.03 토, 1d)
  - [논문 스터디 자료](https://github.com/WannaBeSuperteur/AI-study/blob/main/Paper%20Study/Vision%20Model/%5B2025.05.03%5D%20Art%20Creation%20with%20Multi-Conditional%20StyleGANs.md)
- [x] StyleGAN Discriminator 구조 변경 (05.03 토, 1d)
  - [참고 링크](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_04_08_OhLoRA/stylegan_and_segmentation/README.md#3-3-cnn-model-%EB%82%98%EB%A8%B8%EC%A7%80-%ED%95%B5%EC%8B%AC-%EC%86%8D%EC%84%B1-%EA%B0%92-7%EA%B0%9C)
- [x] train log 에 핵심 속성 값 (eyes, mouth, pose) 의 corr-coef, MAE 추가 (05.04 일, 1d)
- [x] StyleGAN Discriminator 에 StyleGAN-FineTune-v2 CNN 모델의 가중치를 먼저 주입 후 학습 (05.04 일, 1d)
  - 가중치 주입 시, eyes / mouth / pose 속성 CNN 부분의 최종 출력값 (= 예측 속성값) 의 정합성 확인
- [x] Loss 에 따라 StyleGAN Discriminator 를 여러 번 연속 학습하는 메커니즘 적용 시도 (05.04 일, 1d)
- [x] StyleGAN Discriminator 에 Pre-trained Gender CNN 구조 추가 및 그 가중치 주입 (05.04 일, 1d)
  - 이미지 전체를 보는 CNN 구조 추가
  - Pre-trained Gender CNN 에 가중치로 저장된 성별 관련 지식을 추가 이용하면, 유의미한 feature 를 추가로 얻을 수 있을 것으로 판단
- [x] StyleGAN 구조 추가 변경 및 실험 (탐독한 논문 참고) (05.04 일 - 05.06 화, 3d)

## Detailed Model Training Plan
- https://github.com/WannaBeSuperteur/AI_Projects/issues/1#issuecomment-2849092408

## Final Result
- StyleGAN 구조 추가 변경 및 실험했음에도, 성능은 어느 정도 개선되었으나 여전히 불만족스러운 결과 도출
- [상세 실험 결과](https://github.com/WannaBeSuperteur/AI_Projects/blob/main/2025_05_02_OhLoRA_v2/stylegan/stylegan_finetune_v5/train_result.md)

**Comment by WannaBeSuperteur** [comment]
* StyleGAN Discriminator 에 StyleGAN-FineTune-v2 CNN 모델의 가중치를 먼저 주입 후, **해당 가중치는 Freeze** 시키고 학습?
* 그래도 실패 시, 원래 StyleGAN Discriminator 로 원상 복귀?

**Comment by WannaBeSuperteur** [comment]
* 최후의 방법 : 직접 학습하는 대신 [Transformation vector 추출](https://github.com/WannaBeSuperteur/AI-study/blob/main/Paper%20Study/Vision%20Model/%5B2025.05.03%5D%20Art%20Creation%20with%20Multi-Conditional%20StyleGANs.md#3-3-condition-based-%EB%B2%A1%ED%84%B0-%EC%97%B0%EC%82%B0) 하는 방법으로?

**Comment by WannaBeSuperteur** [comment]
* StyleGAN Discriminator 에 Pre-trained Gender CNN 구조 추가 및 그 가중치 주입 (05.04 일, 1d)
  * 이미지 전체를 보는 CNN 구조 추가
  * Pre-trained Gender CNN 에 가중치로 저장된 성별 관련 지식을 추가 이용하면, 유의미한 feature 를 추가로 얻을 수 있을 것으로 판단

**Comment by WannaBeSuperteur** [comment]
## 학습 순서
* [ ] StyleGAN Generator, Discriminator 의 모든 레이어를 Freeze 해제 **(진행중)**
* [ ] (실패 시) StyleGAN Discriminator 에 Pre-trained Gender CNN 구조 추가 및 그 가중치 주입
* [ ] (실패 시) Discriminator 구조를 원래 StyleGAN Discriminator 로 원상복귀 (모든 레이어를 Freeze 해제) 
* [ ] (실패 시) StyleGAN 구조 추가 변경 (탐독한 논문 참고)
* [ ] (실패 시) StyleGAN-FineTune-v1 로부터, SVM 등을 이용하여 핵심 속성 값 Transformation 벡터 (latent vector 인 z 값에 대한) 추출

**Comment by WannaBeSuperteur** [comment]
- StyleGAN-VectorFind-v6 진행 최종 결정
- StyleGAN-FineTune-v1 기반으로 [Label 을 나타내는 벡터를 찾는 방법](https://github.com/WannaBeSuperteur/AI-study/blob/main/Paper%20Study/Vision%20Model/%5B2025.05.05%5D%20Semantic%20Hierarchy%20Emerges%20in%20Deep%20Generative%20Representations%20for%20Scene%20Synthesis.md) 으로 진행
